src/farcasterd/runtime.rs

```diff
@@ -1562,26 +1562,33 @@ impl Runtime {
                                 monero_rpc::RpcClient::new(host);
                             let wallet = wallet_client.wallet();
                             let options = monero_rpc::TransferOptions::default();
-                            match wallet
-                                .transfer(
-                                    [(address, amount)].iter().cloned().collect(),
-                                    monero_rpc::TransferPriority::Default,
-                                    options.clone(),
-                                )
-                                .await
-                            {
-                                Ok(tx) => {
-                                    info!(
-                                        "{} | Auto-funded Monero with txid: {}",
-                                        &swap_id.bright_blue_italic(),
-                                        tx.tx_hash.to_string()
-                                    );
-                                    self.funding_xmr.insert(swap_id, (address, amount, true));
-                                }
-                                Err(err) => {
-                                    warn!("{}", err);
-                                    error!("{} | Auto-funding Monero transaction failed, pushing to cli, use `swap-cli needs-funding Monero` to retrieve address and amount", &swap_id.bright_blue_italic());
-                                    self.funding_xmr.insert(swap_id, (address, amount, false));
+                            for retries in (0..10).rev() {
+                                match wallet
+                                    .transfer(
+                                        [(address, amount)].iter().cloned().collect(),
+                                        monero_rpc::TransferPriority::Default,
+                                        options.clone(),
+                                    )
+                                    .await
+                                {
+                                    Ok(tx) => {
+                                        info!(
+                                            "{} | Auto-funded Monero with txid: {}",
+                                            &swap_id.bright_blue_italic(),
+                                            tx.tx_hash.to_string()
+                                        );
+                                        self.funding_xmr.insert(swap_id, (address, amount, true));
+                                        break;
+                                    }
+                                    Err(err) => {
+                                        if err.to_string().contains("not enough money") || retries == 0 {
```

Comment:
this doesn't catch "not enough unlocked money", which is futile to reattempt prior to a block height increase.

so I'd either
- add a match for this (catchall `not enough`, or `not enough money` || `not enough unlocked money`, or negated `no connection to daemon`) or
- handle `not enough unlocked money` in a separate conditional which inserts to `self.funding_xmr` with the autofund bool = true (to avoid potential double-funding via a manual attempt) and delays handling insufficient unlocked money by reattempting upon a block height increase

Reply:
resolved in 4b32ea40a540fc32ab97023a84841b80c2c3d0e6